BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managed/LifecycleHelper.java

```diff
@@ -213,25 +213,39 @@ public void applyCurrentConfigs() throws AgentServiceNotifyException {
       }
 
       for (BaragonServiceState serviceState : getGlobalStateWithRetry()) {
-        if (!(serviceState.getService().getLoadBalancerGroups() == null) && serviceState.getService().getLoadBalancerGroups().contains(configuration.getLoadBalancerConfiguration().getName())) {
+        if ((serviceState.getService().getLoadBalancerGroups() != null) && (serviceState.getService().getLoadBalancerGroups().contains(configuration.getLoadBalancerConfiguration().getName()))) {
           todo.add(new BootstrapFileChecker(configHelper, serviceState, now));
         }
       }
 
       LOG.info("Going to apply {} services...", todo.size());
 
       try {
+        List<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>> toApply = new ArrayList<>();
         List<Future<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>>> applied = executorService.invokeAll(todo);
         for (Future<Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>>> serviceFuture : applied) {
           Optional<Pair<ServiceContext, Collection<BaragonConfigFile>>> maybeToApply = serviceFuture.get();
           if (maybeToApply.isPresent()) {
-            try {
-              configHelper.bootstrapApply(maybeToApply.get().getKey(), maybeToApply.get().getValue());
-            } catch (Exception e) {
-              LOG.error(String.format("Caught exception while applying %s during bootstrap", maybeToApply.get().getKey().getService().getServiceId()), e);
-            }
+            toApply.add(maybeToApply);
           }
         }
+
+        toApply.parallelStream().forEach(item -> {
```

Comment:
We should performance test parallelStream here a bit. Same for the one in `bootstrapApplyCheck`. There is actually a tipping point where the additional work involved in spinning up the threads and joining them outweighs the benefit of doing the work in parallel, making parallelStream actually slower than stream. We found this example cropping up a bunch in Singularity when offer processing counts got higher than the hundreds for example. parallelStream could still be fine here, just worth testing a bit 👍 

The alternative options to the parallelStream here are a regular stream, or something like the AsyncSempahore we use in Singularity [example](https://github.com/HubSpot/Singularity/blob/5342de1b411fe5513e5c89aa38b5e9b3522746a2/SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosStatusUpdateHandler.java#L314).